### PR TITLE
Add filters on cases view

### LINKED ISSUE #1185: add filter to Cases view
AS a **supervisor** or **admin**,
I WANT TO filter `casa_cases` by a select criteria,
SO THAT I can easily view `casa_cases` that meet that criteria.

### How it works now
`casa_cases` display in numerical order, and are only sortable by ascending and descending order within each column.

### How it should work
`casa_cases` should be filterable by the following:
- active / inactive (see #1182) 
- assigned / not assigned to a volunteer
- assigned to more than one volunteer
- transition aged youth / not transition aged youth
- CINA vs TPR (this shows up in the case number: some start with `CINA` and some start with `TPR`)

### Screenshots
This is how filters work in Volunteer View:
Screen Shot 2020-10-24 at 6 25 28 PM

### PR DESCRIPTION
**Pull request by isaacm**
### What github issue is this PR for, if any?
Resolves #1185

### What changed, and why?
- Implemented all requirements listed in #1185
- _As a bonus: the column picker to toggle show/hide was also implemented to similar to volunteers view_

### How will this affect user permissions?
- Volunteer permissions: N/A
- Supervisor permissions: N/A
- Admin permissions: N/A

### How is this tested? (please write tests!) 💖💪
- Added specs that rely on capybara to toggle UI elements changing filters

### CODE REVIEW

File: app/decorators/casa_case_decorator.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
 class CasaCaseDecorator < Draper::Decorator
   delegate_all
 
+  def status
+    return "Active" if object.active
+
+    "Inactive"
+  end
```
Comment (littleforest) [inline app/decorators/casa_case_decorator.rb:8]: I think the ternary operator might be a little nicer here:

```
def status
  object.active ? "Active" : "Inactive"
end
```

Comment (isaacm) [reply app/decorators/casa_case_decorator.rb:6]: Fixed

**Review by compwron — COMMENTED** [review]
It's looking good! Please add tests and then I think it's ready :)

**Comment by isaacm** [comment]
> It's looking good! Please add tests and then I think it's ready :)

Okay, I'm currently working on tests/screenshots and will update the PR soon (ETA next couple of days).

**Review by compwron — APPROVED** [review]
It's looking good! The build says there's one failing test. (the red x) Please fix it and then let's merge! :)
```
Failures:
  1) volunteer views dashboard sees all their casa cases
     Failure/Error: expect(page).to have_text(casa_case_1.case_number)
       expected to find text "SLAVA-1" in "My Cases\nCase Contacts\nEmancipation Checklist\nLocation\nCASA Org 534\nRole\nVolunteer\nEdit Profile\nGenerate Court Reports\nReport a site issue\nemail206@example.com\nLog out\nMy Cases Pick displayed columns\nFilter by:\nStatus\nAssigned to Volunteer\nAssigned to more than 1 Volunteer\nAssigned to Transition Aged Youth\nCasa Case Prefix\n\nShow\n10\n25\n50\n100\nentries\nSearch:\nCase Number Hearing Type Judge Status Transition Aged Youth Assigned To Actions\nNo matching records found\nShowing 0 to 0 of 0 entries (filtered from 2 total entries)\nPreviousNext"
     
     [Screenshot]: /home/travis/build/rubyforgood/casa/tmp/screenshots/failures_r_spec_example_groups_volunteer_views_dashboard_sees_all_their_casa_cases_490.png
```

**Comment by isaacm** [comment]
For the requirement:
`- CINA vs TPR (this shows up in the case number: some start with CINA and some start with TPR)`

Is it possible to have prefixes other than the two listed in #1185 ?
If so, what is the expected behavior when the prefix for a case number is neither CINA nor TPR?

**Comment by compwron** [comment]
@isaacm For now let's leave the `case_number` field as freetext even though we only expect those two items

**Review by compwron — APPROVED** [review]
:)

**Comment by isaacm** [comment]
Okay sounds good, I'll watch the repo to see if there's any follow-up needed.

I'm glad to contribute and thanks for clearing up  directions.

**Comment by compwron** [comment]
You are invited to the CASA asynchronous retrospective-and-futurespective! Please take a look at https://bit.ly/casaretro (no login required) and add at least one note! Even if it's just "I was here and you are cool" or "omg y'all why do you keep messaging me" we want to hear it. <3
Next week Wednesday 18 November @ 6pm Pacific time we will Zoom about the board and talk about themes and takeaways. (Zoom link in our calendar at http://bit.ly/casacal or slack #casa channel https://rubyforgood.herokuapp.com/ )
Please ESPECIALLY add lots of notes if you won't be able to make it on Wednesday to talk!
* Note: the board will be monitored for spam; message me or any team lead if you see a problem
https://bit.ly/casaretro

**Pull request merged**